### DISCUSSION TITLE
This vcpkg distribution does not have a classic mode instance

### DISCUSSION THREAD

**Discussion by juandent** [microsoft/vcpkg#35403] [General]
Until recent, I was used to run vcpkg in classic mode -- the only mode available. But now there is the manifest mode and it seems the classic mode is not supported any more. Is this correct or is there a way to new make vcpkg to run in classic??

**Comment by AugP** [comment]
vcpkg supports both classic mode and manifest mode, unless you are using the copy that comes with Visual Studio, which is manifest-mode only. That copy is restricted because classic mode changes files in the vcpkg installation directory when dependencies are added, updated, or removed, and the VS installer doesn't like that since it's installed under C:\Program Files\Visual Studio. But you can always install a separate copy of vcpkg and use that. 

To run in classic mode, you just need to run an appropriate command, like "vcpkg install <port>", "vcpkg upgrade", "vcpkg remove <package>", etc. Here is a tutorial for installing dependencies in classic mode: https://learn.microsoft.com/en-us/vcpkg/consume/classic-mode?tabs=msbuild%2Cbuild-MSBuild

Let me know if you have any other questions.

**Comment by juandent** [comment]
```C++
C:\src>vcpkg search sqlite
warning: The vcpkg C:\Program Files\Microsoft Visual Studio\2022\Community\VC\vcpkg\vcpkg.exe is using detected vcpkg root C:\Program Files\Microsoft Visual Studio\2022\Community\VC\vcpkg and ignoring mismatched VCPKG_ROOT environment value C:\src\vcpkg. To suppress this message, unset the environment variable or use the --vcpkg-root command line switch.
error: this vcpkg instance requires a manifest with a specified baseline in order to interact with ports. Please add 'builtin-baseline' to the manifest or add a 'vcpkg-configuration.json' that redefines the default registry.
```

I set VCPKG_ROOT to my local instalation (c:\src\vcpkg) and adding this installation to the PATH variable.


One way to avoid this is to fully qualify the local vcpkg installation correct?

**Comment by StarGate-One** [comment]
My advice if you only use vcpkg in classic mode:
Uninstall the vcpkg from Visual Studio 2022 (as it is crippled and uses only manifest mode and the cli is disabled) via the Visual Studio installer - go to the `Individual Components` tab, enter `vcpkg` into the search window, hit enter and uncheck the `vcpkg package manager` component.
Click `close` button and let the visual studio installer remove the component.
Exit the visual studio installer.
Clone the vcpkg from repo into the `C:\src\`.
`cd C:\src\vcpkg`.
Run the bootstrap `.\bootstrap-vcpkg.bat` in `C:\src\vcpkg`.
Run `vcpkg integrate install` from the same `C\src\vcpkg` (now Visual Studio will find all the installed vcpkg components).
Run `vcpkg install <ports/features>`, `vcpkg remove <ports/features>` and/or `vcpkg upgrade` as needed.
